### ISSUE TITLE
[QSRlib] Implement user custom config file

### ISSUE THREAD

**Issue by pet1330** [strands-project/strands_qsr_lib#45]
As described in [this](https://github.com/strands-project/strands_qsr_lib/pull/43#issuecomment-101237270) comment. The ability to pass QSR arguments using a user defined config file.

**Comment by yianni** [comment]
My idea for this would be that there is one "default" ini file where appropriate QSRs will read from their sections their ini settings. Something like:
```ini
[rcc3]
some_setting_1 = 2.3
some_setting_2 = 3.4

[qtcab]
quantization_factor = 1.1
```

This can be overwritten on the request/constructor by passing a user defined path/filename ini file. The example client could the run as:
```bash
example_ros_client qtcab --ini /home/foo/my_qtcab_settings.ini
```

@pet1330 @cdondrup Thoughts? As it will most certainly require changing somethings in the QSRs themselves I would think.

**Comment by pet1330** [comment]
Sounds good, I think the ability to set parameters per request should be kept and that if parameters are set, then they should be used over the ones defined in the config file.

Also would it be worth defining a return type at the bottom? Allowing people to request the relations as a string, dictionary , list, etc...

**Comment by cdondrup** [comment]
Sounds good. I will still mainly use dynamic parameters, to be able to change them at run time but having default ones sounds great. Regarding the return type, let's stick with strings and leave it to the client to parse them into the format they need. If you change them using a config file, not all client programmes might be aware and will therefore fail because they expect a different type. I would be careful with something like that.

**Comment by yianni** [comment]
I think that both cases are possible (initialise from file and pass/change at runtime via the request).

With respect to the return type, my thoughts were what Christian described above. Both developers and users will expect a particular type and anything else will create problems. There might be functions provided that change the type but I am not really keen on promoting them. In fact I would probably even leave it totally to the client side as if this happens then it means that is an exception and they will know better what format is needed.

**Comment by yianni** [comment]
see #49

**Comment by yianni** [comment]
closed by #49
